src/compute/src/compute_state.rs

```diff
@@ -141,6 +141,22 @@ pub struct ComputeState {
     /// dataflow. Multiple collections can reference the same token if they are exported by the
     /// same dataflow.
     suspended_collections: BTreeMap<GlobalId, Rc<dyn Any>>,
+
+    /// When this replica/cluster is in read-only mode it must not affect any
+    /// changes to external state. This flag can only be changed by a
+    /// [ComputeCommand].
+    ///
+    /// Everything running on this replica/cluster must obey this flag. At the
+    /// time of writing the only part that is doing this is `persist_sink`.
+    ///
+    /// NOTE: In the future, we might want a more complicated flag, something
+    /// that allows awaiting a change, or something that tells us after which
+    /// timestamp we are allowed to write. In this first version we are keeping
+    /// things as simple as possible!
+    pub read_only_rx: watch::Receiver<bool>,
```

Comment:
> something that allows awaiting a change

Awaiting a change is possible with a `watch`, no?

Reply:
yeah, that was from before I changed it to a `watch`, fixing 👌